process/factory/containers/preProcessorsContainer.go

```diff
@@ -86,3 +86,17 @@ func (ppc *PreProcessorsContainer) Remove(key block.Type) {
 func (ppc *PreProcessorsContainer) Len() int {
 	return ppc.objects.Len()
 }
+
+func (ppc *PreProcessorsContainer) Keys() []block.Type {
```

Comment:
There are repeating functions on different containers having `objects *hashmap.HashMap`. Can't we have a common implementation? Maybe use struct embedding to reuse the code

Reply:
it is a container for a certain type, for a specific interface. One general would return interface{} and you would need type assertions for every get.

Reply:
`func (ppc *PreProcessorsContainer) Keys() []block.Type` returns `[]block.Type` just as the one in the `intermediateTransactionHandlersContainer.go`, maybe I am missing something :)

Reply:
Keys return the same thing. but 60% of the public methods return different elements Preprocessor vs IntermediateProcessor. Get(key block.Type) (PreProcessor, error)